Update To Governance and Contributing Guidelines (#1583)
pyscript/pyscript: GOVERNANCE.md

> **1.1. Maintainers**. Maintainers are responsible for organizing activities around developing, maintaining, and updating the Project. Maintainers are also responsible for determining consensus. This Project may add or remove Maintainers with the approval of the current Maintainers.

Reviewer: Please add:

```
Maintainers are defined as GitHub users listed as maintainers in the main PyScript repository.
```

The PyScript project implements a mixed committee/BDFL (Benevolent Dictator for Life) decision making model, where the BDFL works with the committee with the ultimate goal of advancing the project while developing its vision.

**3.1.1. Rationale**

_Why this model rather than any other ?_

It comes down to “vision” and consistency.

Given the current status and life of the PyScript project, we believe that the best way to encourage a mixed approach where the active `maintainers` group acts as a committee, effectively being at the center of the decision making process, and the BDFL act as ultimate gate keeper of the vision and the project users.

**2.1. Consensus-Based Decision Making**. The project standard way of making decisions is through consensus of the Maintainers and the automatic blessing of the BDFL. While explicit agreement of all Maintainers is preferred, it is not required for consensus. Rather, the Maintainers will determine consensus based on their good faith considering a number of factors, including the dominant view of the Contributors and nature of support and objections. The Maintainers will document evidence of consensus in accordance with these requirements.

> In cases of controversial discussions and on exceptional cases, the BDFL has the right (and duty) to make a call to confirm or decline a specific decision. This right is to be exercised very carefully, and used only on delicate matters whenever it is fit.

Reviewer: ```
In cases of controversial discussions and in exceptional cases,...
```

**2.2. Appeal Process**. Decisions may be appealed by opening an issue and that appeal will be considered by the Maintainers in good faith, who will respond in writing within a reasonable time.

**2.3. BDFL Role**

The main goal of the PyScript BDFL is to serve as support in the decision making process, and provide a compass to help guide the project towards its vision and its core principles. It is not meant to replace the group of maintainers in the decision process nor to discourage healthy conversation and proposals or overrule the voting process.

The BDFL has the following responsibilities:

-   Never use their rights in ways that discourage an open and respectful dialogue among Maintainers, Contributors, and Users.
-   Ensure a fair and respectful voting process.

and rights:

> -   Right to bypass the voting process towards a decision in case there isn't an unanimous consensus on a specific matter among all the other Maintainers of the project or in case they consider the decision against the project vision or best interest.

Reviewer: Please change this to (so we have the conditional explicit at the start):

```
-  If the maintainers cannot or will not make a decision through their usual processes, the BDFL has the right to bypass such processes and be the final arbiter of the decision in hand.
```

I'm not comfortable with the bit, "in case they consider the decision against the project vision or best interest". That's way too broad and basically gives BDFL `sudo` powers by fiat. Rather, the version I've written ensures the BDFL is *always* _only_ the arbiter of last resort when there's a breakdown in the usual maintainer's process.

It's the difference between a parliamentary monarchy and absolute monarchy. In the first, the monarch is not absolute ruler, but a useful "last resort" and bureaucratic function. In the second, they can do as they please (never a good situation because there needs to be guards).

Author: @ntoll  As I expected, that's probably the point that requires more discussion. 

I've added the paragraph the way it is because I wanted to make sure that this BDFL model is more like a soft BDFL model where the maintainers can still overrule a BDFL decision (through a unanimous vote). That is not how usual BDFL models work. Funny enough, all the other feedback I've got so far was in the opposite direction 🤣 

In short, I've got 2 types of feedback so far (below is my interpretation... may be incorrect): 

* This sounds more like a weak and doesn't really solve the problem we have around converging toward a decision effectively. We should have the **rule** that the BDFL can bypass any vote, even unanimous, and have the ** practice ** that this rule is exercised only in very rare cases 
* This sounds like a monarchy. We don't want a BDFL but a person to resolve impasses within maintainers.

I'm open to both options (the goal here is not the proposal itself but how it can help the project achieve its vision). The only issue with your proposed text is that it's way too loose to move any discussion forward because `cannot or will not make a decision through their usual processes` doesn't define how we agree on what it means. Is this after a day? A week? What are the conditions where we say "maintainers cannot or will not make a decision through their usual processes"?

Reviewer: Offered in a constructive spirit:

So, of the two bullet points, the first is definitely tyranny: one person can do whatever they like with no recourse to holding them to account. This is not a good thing and I feel deeply uncomfortable that we're even considering it. I'm not sure what I would do were we to adopt this way of working because it goes against everything I believe in. 

The second is more like what you get in the US system or the UK's parliamentary monarchy, where there's a significant person with a casting vote (the VP in the case of votes on the Hill, or the royal prerogative here in the UK). They act as the super powerful unblocker when the usual processes don't work. But the point is, their power is specific, constrained and managed. They complement (rather than replace) a system based upon collective ownership and decision making.

If there's one thing history tells us, "the practice is..." sounds well and good, until the BDFL decides otherwise, including changing all these rules. That's why the BDFL should have a very powerful but well defined and specific role in unblocking decision making (and only that). It's all about scope of influence. Just remember, in the future the "B" bit of BDFL is not guaranteed - and history (again and again and again) shows us this to be the case, including many open source projects that imploded because "I'm in charge" is not a great way to manage disputes.

FWIW my preference is for something like the current CPython situation: a steering council, elected by maintainers, containing an odd number (I suggest 3), who arbitrate and decide by simple majority when consensus cannot be reached by the maintainers. It's mature, sustainable and representative. Since it's a council, it also removes making it all about an individual personality while ensuring the heavy burden of responsibility is shared. It's also just as lightweight as having a BDFL too. As Winston Churchill said )paraphrasing from memory), "democracy is the very worst form of governance, except for all the others that have been tried".

Anyway... I have a very strong dislike of tyranny for deeply personal moral and ethical reasons and I feel very uncomfortable about it. If we must have a BDFL - their role needs to be clear, constrained and with an escape hatch so they can be removed.

Put another way, an unconstrained BDFL falls short when answering these famous five questions concerning power:

* What power have you got?
* Where did you get it from?
* In whose interest do you exercise it?
* To whom are you accountable?
* How can we get rid of you?

They can only answer in the follow way:

* Unlimited power (I'm a BDFL).
* Because I'm BDFL with unlimited power.
* My own - that's the point... I exercise my judgement for what *I see* is the good of the project, and you just have to lump it.
* Nobody.
* You can't (or you walk away / fork the project / watch it implode etc...).

I hope you agree, none of these answers are the sort we want to give to anyone wanting to join our project. Especially when the only recourse to go against the BDFL is to walk away or fork the project. Our community is littered with projects that imploded because the BDFL system alienated people rather than an alternative way of working that empowered, engaged and nurtured them.

And to those who say, "but it's only a software project, we just want someone to make decisions so I can get on with the important stuff of writing code" ~ **writing software *is* a political activity** because we're organising and exercising control of resources, processes and how things unfold _through and in a digital medium_. This is why it's so important to get our own internal "political" structures well defined: _because asking what they should be tells us what sort of project we'll become_.

I don't want to work in a tyranny... no matter how nice the tyrant may be. History tells us it never ends well for all concerned, especially if the project becomes very successful (as we hope will be the case for PyScript).

Peace. :hugs:

Author: I'd like to address a quick thing first (because it puts this conversation under very bad and emotionally overloaded terms). 

> So, of the two bullet points, the first is definitely tyranny

That is simply not true. Tyranny and Dictatorship are 2 different words with 2 different meanings. A dictator is someone who acquires power [without the consent of the people] (in this and most cases in sw that's not true) and is (or can be) aided by a group of counselors. A Tyrant is someone who leads in an oppressive and ruthless way, in their own interest and above the interest of their community as a whole. I hope the difference between a Benevolent Dictator vs a Tyrant is very clear to everyone (and on a personal note, I do abhor the idea that anyone close to the PyScript project could be a Tyrant if elected a BDFL). 

With that out of the way, I think we are not that far apart in how we see things. Let me start getting something out there right away: I'm not advocating for a BDFL with unlimited powers. If I wanted, it was very easy to start that right away when I created the project 😉 .  Even though it's not my personal preference, I also don't think it's a bad thing per se. There are plenty of examples of software projects that work that way and they work great. (And are actually way closer to a democracy than a dictatorship, not to mention a tyranny).

I do have to acknowledge that consensus among the maintainers, like we are running things today, is not being very effective and we've been having a hard time/moving slowly in the decision-making process. Tbh, I don't think governance models are a one size fits all thing. There are BDFL-led projects that ran well with that model (and then not so great when they turned into a steering council mode) and many great projects that work well with a steering council model.

Now, going back to a point you made: **writing software is a political activity**. I usually say "Writing software is a social activity" (because politics kind of have a smell to it 😃  and I think people > politics as a whole) but we are not far off. In fact:

* the reason I wrote `Right to bypass the voting process towards a decision in case there isn't a unanimous consensus on a specific matter among all the other Maintainers of the project` was because I did want to create a mechanism for the steering council (the maintainers in this case) to limit the BDFL power
* I personally prefer a model with a steering council led by a leader (similar to a board and a chairman of the board) that has some extra rights to allow this person to [try to] avoid deadlocks, slowdowns, bureaucracy overhead, etc..  while being the compass
* While the D-word (dictator) often implies "without consent of the people", it hasn't been the case in most cases I've seen in regard to a BDFL in software
* I'm opposed to this role to created without people's consent.
* I actually was going to propose a "soft-BDFL" model where it's really more like a Chairman of a Steering council that has similar rights to what I proposed but can be voted down by the council and can only bypass votes below a certain threshold
* I did not write that proposal here because, actually, most of the feedback I had when talking with other maintainers was that they preferred a more classic BDFL role (so I wrote what I think the majority prefers)

Now, I'll repeat my preference here just to make sure it's added to the options and called out explicitly: "soft-BDFL" model (or maybe we should call it a steering council with a Chairperson) where the Chair has similar rights to what I proposed but can be voted down by the council and can only bypass votes below a certain threshold. This is not about Tyranny (dang, I can't help but hate this term) but about a group of people putting trust in a leadership group and a leader.

(Also think this is a hard conversation over text)

## 3. How We Work.

### 3.1. Principles

> **3.1.1. Openness**. Participation is open to anyone who is directly and materially affected by the activity in question. There shall be no undue financial barriers to participation. In that sense, discussions should prioritize public channels (Github discussions/issues/PRs or Discord) whenever possible.

Reviewer 2: Maybe some comment here also about record keeping and discoverability of past decision making. I think discord is inferior to github for technical discussions for this reason: it's going to be hard for anyone to locate the discussions later. Of course there are tradeoffs here, for instance having a video meeting is very efficient for technical discussions but it is often not very inclusive and if nobody takes meeting minutes then there will be no record of what happened.

> **3.1.2. Balance**. The development process should balance the interests of Contributors and other stakeholders. Contributors from diverse interest categories shall be sought with the objective of achieving balance.

Reviewer 2: Do we have a definition of "Contributor"? I think it would be good to start this document with a definition, preferably an expansive one (for instance recognizing engaging in public discussion around the project as a form of contribution).

### 3.2.1. Steps Towards Consensus

**Informal consensus seeking.** Decision making starts with informal consensus seeking through discussion. The goal of this phase is to refine the proposal, consider alternatives, weigh trade-offs, and attempt to find informal consensus. The legitimacy of the consensus-seeking process is predicated on all stakeholders having their voices heard, so `maintainers` must be proactive in providing opportunities for all relevant stakeholders to provide input. If `maintainers` at informal consensus, they may immediately move to document and enact the decision. This is the consensus-seeking phase.

**Calling a vote.** While "calling a vote" seems like a slow and bureaucratic process, its goal is to ensure that all stakeholders had a say in the process of making a decision. In an effort to keep things simple and lean, this step can be considered "integrated" into the previous (informal) step of Informal Consensus Seeking if there's enough support and agreement so that is clear that the proposal has the support from the majority of `maintainers`.

> If that's not the case and two or more parties are not aligned, any `maintainer` can call the matter to a vote. When that member (the sponsor) calls the vote, they shall summarize the proposal in its current to make sure the proposal is clear to everyone. After the proposal is seconded by at least another `maintainer`, voters have 4 days to vote. `Maintainers` may consider longer voting periods as necessary for special circumstances, or shorter periods only if all voting members are present. The decision will be determined by a simple majority of non-blank votes for binary decisions (i.e., approving a proposal) and ranked choice for multi-class decisions (one among many, or several among many). The sponsor may update the proposal at any point during the voting period, in which case the voting period will be reset.

Reviewer 3: ```suggestion
If that's not the case and two or more parties are not aligned, any `maintainer` can call the matter to a vote. When that member (the sponsor) calls the vote, they shall summarize the proposal in its current form to make sure the proposal is clear to everyone. The summary and (and ensuing vote) shall be posted in a permanent public forum, with GitHub Discussions or Issues being the default preference.

After the proposal is seconded by at least another `maintainer`, voters have 4 days to vote. `Maintainers` may consider longer voting periods as necessary for special circumstances, or shorter periods only if all voting members are present. The decision will be determined by a simple majority of non-blank votes for binary decisions (i.e., approving a proposal) and ranked choice for multi-class decisions (one among many, or several among many). The sponsor may update the proposal at any point during the voting period, in which case the voting period will be reset.
```

NOTE: While only `maintainers` can vote, anyone can comment and participate to the discussion around the proposal, in the spiriti of opennes and inclusivity at the base of this project.

**Voter participation and quorum.** There is no minimum quorum for a vote to be considered valid, unless a voter had expressed their inability to vote and asked for an extended voting time window. All votes must include a "abstinent" option, for voters to express their desire to be neutral to decision while still casting a vote.

> All `maintainers` are highly encouraged to participate and exercise their rights expressing their opinion in at least 2/3 of formal votes per calendar year. Members that have not met the 2/3 vote participation threshold for a year and have a low contribution rate in that year may be to step down from their `maintainer` status at the end of that year. Those individuals remain eligible to rejoin the `maintainers` group in the future as they become available to participate at the required level. The quorum for all formal votes will be 50% and a “blank” option will always be included, with the “blank” option counting towards the quorum but not included in totals for calculating results.

Reviewer 3: ```suggestion
All `maintainers` are highly encouraged to participate and exercise their rights expressing their opinion in at least 2/3 of formal votes per calendar year. Members that have not met the 2/3 vote participation threshold for a year and have a low contribution rate in that year may be asked to step down from their `maintainer` status at the end of that year. Those individuals remain eligible to rejoin the `maintainers` group in the future as they become available to participate at the required level. The quorum for all formal votes will be 50% and a “blank” option will always be included, with the “blank” option counting towards the quorum but not included in totals for calculating results.
```

**Recording.** Once a decision has been made during the consensus-seeking phase or by a formal vote, the initiative `sponsor` will record the decision in the related discussion, for posterity.

> ### 3.2.2. Things to avoid

Reviewer 3: ```suggestion
### 3.2.2. Keys to Successful Consensus
```

A few of the things in this section are "don't do this" and a few are "remember to do this", so I dropped in some proposed language making them all things _to do_. Just a style thing, feel free to take or leave.

Author: let's be affermative! :)

> **Rushing a decision**: While we want prioritize a lean and fast process, we also want to make sure that everyone express their opinions.

Reviewer 3: ```suggestion
**Allow Time for Decisions**: While we want prioritize a lean and fast process, we also want to make sure that everyone express their opinions. The process should never feel "rushed."
```

> **Calling for a vote too soon**: Expanding on the item above, `maintainers` should not call a vote to short-circuit an ongoing discussion that is still productive in terms of exploring ideas and feedback. Votes should be called only when discussion has explored the space, stakeholders have provided input and consensus hasn't been reached through the informal process

Reviewer 3: ```suggestion
**Use Voting at the End of the Discussion**: Expanding on the item above, formal votes should be called only when discussion has explored the space, stakeholders have provided input and consensus hasn't been reached through the informal process. `Maintainers` should be should not call a vote to short-circuit an ongoing discussion that is still productive in terms of exploring ideas and feedback. 
```

**Give the right weight to decisions** `Maintainers` and those proposing decisions should explicitly distinguish between decisions that are two-way (easy to reverse later) and one-way (difficult or impossible to reverse later). For one-way doors, `maintainers` should carefully weigh alternatives and tradeoffs and take extra care to ensure broad participation and stakeholder input. For two-way doors, `maintainers` should feel free to move quickly, without compromising the principles and procedures described herein.

> **Lack of data** In the context of the above item and in the interest of making solid decisions, argumente should bring data and examples to back them up. "Gut feelings" will not be considered with the same relevance of arguments backed by data and examples.

Reviewer 3: ```suggestion
**Utilize Data** In the context of the above item and in the interest of making solid decisions, arguments should bring data and examples to back them up. "Gut feelings" will not be considered with the same relevance of arguments backed by data and examples.
```

> 

Reviewer 3: ```suggestion

**Acknowledge Some Questions Don't Have 'Right' Answers**: Expanding on the item above, many proposals or question won't have any objectively 'right' answers (e.g. naming things). `Maintainers` should acknowledge when they are expressing a preference, while avoiding declaring their preferences as "correct." Maintainers should also be wary that these matters can be the most prone to [bikeshedding](https://en.wikipedia.org/wiki/Law_of_triviality) and judge how much time they're spending on these matters accordingly.

```

> **Forgetting how we work** Given how passionate `maintainers` and users in general can be passionate about the project, discussion can get passionate. This should never be an excuse to operate with all our principles in mind and treat each other with respect, empathy and kindness.

Reviewer 3: ```suggestion
**Remember How We Work* Given how passionate `maintainers` and users in general can be passionate about the project, discussion can get passionate. This should never be an excuse to operate with all our principles in mind and treat each other with respect, empathy and kindness.
```

> **Explicit is better than implicit** `Sponsors` and `maintainers` should proactively solicit input from relevant stakeholders and should not assume that silence is consent without attempting to reach out to those individuals.

Reviewer 3: ```suggestion
**Favor Explicit over Implicit** `Sponsors` and `maintainers` should proactively solicit input from relevant stakeholders and should not assume that silence is consent without attempting to reach out to those individuals.
```

> **Planning before acting** If you are interested in a decision being made, it is your responsibility to encourage voting member/stakeholder/community participation in the decision-making process. If you cannot get such participation, you may want to hold off on doing any significant work on the matter.

Reviewer 3: ```suggestion
**Plan before acting** If you are interested in a decision being made, it is your responsibility to encourage voting member/stakeholder/community participation in the decision-making process. If you cannot get such participation, you may want to hold off on doing any significant work on the matter.
```

## 3. How We Work.

### 3.1. Principles

**3.1.1. Openness**. Participation is open to anyone who is directly and materially affected by the activity in question. There shall be no undue financial barriers to participation. In that sense, discussions should prioritize public channels (Github discussions/issues/PRs or Discord) whenever possible.

**3.1.2. Balance**. The development process should balance the interests of Contributors and other stakeholders. Contributors from diverse interest categories shall be sought with the objective of achieving balance.

**3.1.3. Coordination and Harmonization**. Good faith efforts shall be made to resolve potential conflicts or incompatibility between releases in this Project.

**3.1.4. Consideration of Views and Objections**. Prompt consideration shall be given to the written views and objections of all Contributors.

**3.1.5. Written procedures**. This governance document and other materials documenting this project's development process shall be available to any interested person.

### 3.2. Decision-Making Guide

> This section describes how PyScript governing bodies (generally called “`maintainers`”) make decisions. This section provides clarity to all users, around the process around how decisions are made while providing a clear guide that can be consulted in case of doubts and the process to resolve conflicts and reach a resolution.

Reviewer 2: Maybe:
```suggestion
This section describes how the PyScript maintainers make decisions. This section is intended to:
* provide a decision-making process for maintainers
* provide clarity to users about the maintainers' decision-making process  
* provide a process for disagreeing and for resolving disagreements
```

> We seek to honor the principles of collaboration, inclusive participation, and responsive decision making. Some aspects of this decision-making guide are required while others are provided as recommendations and are optional. We have clearly noted the optional aspects of decision making below.

Reviewer 2: ```suggestion
We seek to honor the principles of collaboration, inclusive participation, and responsive decision making. Some aspects of this decision-making guide are mandatory while others are merely recommended. We have clearly noted the recommendations.
```

> Finally, the PyScript `maintainters` group may either intervene, or be called upon, in the event that issues arise in the decision-making process and it comes short in providing the right guidance on specific subjects. This includes, but is not limited to: process ambiguity, violations of the decision-making process, mitigating circumstances that require process exceptions, etc.

Reviewer 2: ```suggestion
In the event that issues arise in the decision-making process that cannot be resolved by the processes in this guide, the maintainers may as a group decide on an alternate process. This includes, but is not limited to: process ambiguity, violations of the decision-making process, mitigating circumstances that require process exceptions, etc.
```
How is this invoked? By majority vote among maintainers? Supermajority? Who can bring a vote? Of course, we can just figure it out as it comes up but it might be worth providing some additional detail here.

NOTE: The decision-making process is intended to balance broad participation of stakeholders with agility. This process aims to be a tool to help and not get in the way of the decision making process itself (turning it into a long and bureaucratic hassle), and is inspired by other successful projects such as the Python and Jupyter projects.

### 3.2.1. Steps Towards Consensus

> **Informal consensus seeking.** Decision making starts with informal consensus seeking through discussion. The goal of this phase is to refine the proposal, consider alternatives, weigh trade-offs, and attempt to find informal consensus. The legitimacy of the consensus-seeking process is predicated on all stakeholders having their voices heard, so `maintainers` must be proactive in providing opportunities for all relevant stakeholders to provide input. If `maintainers` at informal consensus, they may immediately move to document and enact the decision. This is the consensus-seeking phase.

Reviewer 2: ```suggestion
**Informal consensus seeking.** Decision making starts with informal consensus seeking through discussion. The goal of this phase is to refine the proposal, consider alternatives, weigh trade-offs, and attempt to find informal consensus. The legitimacy of the consensus-seeking process is predicated on all stakeholders having their voices heard, so `maintainers` must be proactive in providing opportunities for all relevant stakeholders to provide input. If `maintainers` arrive at an informal consensus, they may immediately move to document and enact the decision.
```

> **Calling a vote.** While "calling a vote" seems like a slow and bureaucratic process, its goal is to ensure that all stakeholders had a say in the process of making a decision. In an effort to keep things simple and lean, this step can be considered "integrated" into the previous (informal) step of Informal Consensus Seeking if there's enough support and agreement so that is clear that the proposal has the support from the majority of `maintainers`.

Reviewer 2: ```suggestion
**Calling a vote.** If an informal consensus cannot be reached, a maintainer may call a vote. While "calling a vote" seems like a slow and bureaucratic process, its goal is to ensure that all stakeholders have a say in the process of making a decision. In some cases, the vote-taking can be integrated into the Informal Consensus Seeking step. However, in cases where the decision is very important or there is strong disagreement, then it can be put to a formal vote.
```
It's a bit unclear here: you first say "all stakeholders"  but then that the proposal only needs support from a "majority of maintainers". Presumably only the maintainers get a vote? But then this is ensuring that all maintainers have a say in the process, not that all stakeholders do.

> If that's not the case and two or more parties are not aligned, any `maintainer` can call the matter to a vote. When that member (the sponsor) calls the vote, they shall summarize the proposal in its current form to make sure the proposal is clear to everyone. The summary and (and ensuing vote) shall be posted in a permanent public forum, with GitHub Discussions or Issues being the default preference.

Reviewer 4: nit: should we say GitHub discussion is, in general, preferred over the forum? I am guilty myself of not being too active in the forum, but surely I'll get a notification if out of a GitHub discussion.

Reviewer 2: Personally, I was not aware that the PyScript forum on the anaconda website existed before reading this.

Author: Yeah. It was active in the first 6 months (before we had GH Discussions) and the dynamics slowly migrated towards GH discussions and Discord. 

I'm +1 on removing that suggestion in favor of both.

After the proposal is seconded by at least another `maintainer`, voters have 4 days to vote. `Maintainers` may consider longer voting periods as necessary for special circumstances, or shorter periods only if all voting members are present. The decision will be determined by a simple majority of non-blank votes for binary decisions (i.e., approving a proposal) and ranked choice for multi-class decisions (one among many, or several among many). The sponsor may update the proposal at any point during the voting period, in which case the voting period will be reset.

> NOTE: While only `maintainers` can vote, anyone can comment and participate to the discussion around the proposal, in the spiriti of opennes and inclusivity at the base of this project.

Reviewer 2: ```suggestion
NOTE: While only `maintainers` can vote, anyone can comment and participate in the discussion around the proposal.
```

**Voter participation and quorum.** There is no minimum quorum for a vote to be considered valid, unless a voter had expressed their inability to vote and asked for an extended voting time window. All votes must include a "abstinent" option, for voters to express their desire to be neutral to decision while still casting a vote.

> All `maintainers` are highly encouraged to participate and exercise their rights expressing their opinion in at least 2/3 of formal votes per calendar year. Members that have not met the 2/3 vote participation threshold for a year and have a low contribution rate in that year may be asked to step down from their `maintainer` status at the end of that year. Those individuals remain eligible to rejoin the `maintainers` group in the future as they become available to participate at the required level. The quorum for all formal votes will be 50% and a “blank” option will always be included, with the “blank” option counting towards the quorum but not included in totals for calculating results.

Reviewer 2: Do we expect there to be very many formal votes?

Author: @hoodmane can you elaborate?

Actually, it'd be good to actually give more details about the process in order to try and make it the easier and leaner as possible

**Recording.** Once a decision has been made during the consensus-seeking phase or by a formal vote, the initiative `sponsor` will record the decision in the related discussion, for posterity.

### 3.2.2. Keys to Successful Consensus

> **Allow Time for Decisions**: While we want prioritize a lean and fast process, we also want to make sure that everyone express their opinions. The process should never feel "rushed."

Reviewer 2: ```suggestion
**Allow Time for Decisions**: While we want prioritize a lean and fast process, we also want to make sure that everyone express their opinions. The process should never feel rushed.
```

**Use Voting at the End of the Discussion**: Expanding on the item above, formal votes should be called only when discussion has explored the space, stakeholders have provided input and consensus hasn't been reached through the informal process. `Maintainers` should be should not call a vote to short-circuit an ongoing discussion that is still productive in terms of exploring ideas and feedback.

**Give the right weight to decisions** `Maintainers` and those proposing decisions should explicitly distinguish between decisions that are two-way (easy to reverse later) and one-way (difficult or impossible to reverse later). For one-way doors, `maintainers` should carefully weigh alternatives and tradeoffs and take extra care to ensure broad participation and stakeholder input. For two-way doors, `maintainers` should feel free to move quickly, without compromising the principles and procedures described herein.

> **Utilize Data** In the context of the above item and in the interest of making solid decisions, arguments should bring data and examples to back them up. "Gut feelings" will not be considered with the same relevance of arguments backed by data and examples.

Reviewer 2: ```suggestion
**Utilize Data** In the context of the above item and in the interest of making solid decisions, arguments should bring data and examples to back them up. "Gut feelings" do not carry as much weight as data and examples.
```

**Acknowledge Some Questions Don't Have 'Right' Answers**: Expanding on the item above, many proposals or question won't have any objectively 'right' answers (e.g. naming things). `Maintainers` should acknowledge when they are expressing a preference, while avoiding declaring their preferences as "correct." Maintainers should also be wary that these matters can be the most prone to [bikeshedding](https://en.wikipedia.org/wiki/Law_of_triviality) and judge how much time they're spending on these matters accordingly.

**Remember How We Work** Given how passionate `maintainers` and users in general can be passionate about the project, discussion can get passionate. This should never be an excuse to not operate with all our principles in mind and make sure we treat each other with respect, empathy and kindness.

**Favor Explicit over Implicit** `Sponsors` and `maintainers` should proactively solicit input from relevant stakeholders and should not assume that silence is consent without attempting to reach out to those individuals.

> **Plan before acting** If you are interested in a decision being made, it is your responsibility to encourage voting member/stakeholder/community participation in the decision-making process. If you cannot get such participation, you may want to hold off on doing any significant work on the matter.

Reviewer 2: If people don't participate, in some cases that could be taken as a sign that most people just don't care? Either the decision won't affect them or they don't understand yet how it will affect them. In the case where it actually won't affect them, it's pretty reasonable to just go ahead with the engaged parties. In the case where they can't meaningfully participate because they don't understand the issues at hand and what their consequences are, then maybe it's important to address the situation in some way, maybe with more explanation.

Author: Yeah, that's a good point. I actually haven't thought of that aspect which is actually even important right now where we are still in the early phases of spreading the word and trying to get people involved. With that in mind, I'd propose to to remove this last sentence for the time being
